### PR TITLE
Get my profile endpoint implementation

### PR DESCRIPTION
**Pull request by Shrenik0321**
## Purpose
Implemented the get my profile endpoint.

## Goals
It sends an array of all table record objects when the api endpoint is called. 

## Approach
Built a simple api which uses the find() function to get all table records and send as a response.

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -13,13 +14,14 @@ initConnection();
...
 app.use(cors());
-app.use(passport.initialize()); 
+app.use(passport.initialize());
 
 app.get('/', (req, res) => {
   res.send('ScholarX Backend');
 });
 
 app.use('/api/auth', authRouter);
+app.use('/api/profile', profileRouter);
```
Comment (anjula-sack) [inline src/app.ts:24]: I made a change to the api: https://api-scholarx.sefglobal.org/api/me/profile

```suggestion
app.use('/api/me', profileRouter);
```

File: src/routes/profile.route.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import express from 'express';
+import { getProfile } from '../controllers/profile.controller';
+
+const profileRouter = express.Router();
+
+profileRouter.get('/me', getProfile);
```
Comment (anjula-sack) [inline src/routes/profile.route.ts:6]:
```suggestion
profileRouter.get('/profile', getProfile);
```

File: src/controllers/profile.controller.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import { dataSource } from '../configs/dbConfig';
+import { Request, Response } from 'express';
+
+export const getProfile = async (req: Request, res: Response) => {
+  try {
+    const profileRepository = dataSource.getRepository(Profile);
+
+    // Find and return an array of all table rows in the profile table.
+    const allProfiles = await profileRepository.find();
+    res.status(200).json({ allProfiles });
```
Comment (anjula-sack) [inline src/controllers/profile.controller.ts:11]: This should return only the logged-in user's profile. The request will contain the authorization header with the jwt token we are getting from the sign-in. That jwt token contains the userId. Using that you can fetch the profile.

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/routes/profile.route.ts (hunk 2 above)
Comment (anjula-sack) [inline src/routes/profile.route.ts:6]: Need to use the passport authentication

```suggestion
profileRouter.get('/profile',  passport.authenticate('jwt', { session: false }), getProfile);
```

**Comment by anjula-sack** [comment]
@Shrenik0321 can you write tests for this?

**Comment by Shrenik0321** [comment]
> @Shrenik0321 can you write tests for this?

Sure. I've not written tests before but I can give it a try.

**Comment by anjula-sack** [comment]
> 

No worries, It's time to learn. We can write tests for the controller and the service.

**Comment by Shrenik0321** [comment]
> > > @Shrenik0321 can you write tests for this?
> > 
> > 
> > Sure. I've not written tests before but I can give it a try.
> 
> No worries, It's time to learn. We can write tests for the controller and the service. @Shrenik0321

Alright, thanks. I'll be able to refer to that an write tests for future prs.

**Comment by anjula-sack** [comment]
> > > > @Shrenik0321 can you write tests for this?
> > > 
> > > 
> > > Sure. I've not written tests before but I can give it a try.
> > 
> > 
> > No worries, It's time to learn. We can write tests for the controller and the service. @Shrenik0321
> 
> Alright, thanks. I'll be able to refer to that an write tests for future prs.

Great, can you write tests for this? Get some help with chatGPT. @Shrenik0321

**Comment by Shrenik0321** [comment]
> > > > > @Shrenik0321 can you write tests for this?
> > > > 
> > > > 
> > > > Sure. I've not written tests before but I can give it a try.
> > > 
> > > 
> > > No worries, It's time to learn. We can write tests for the controller and the service. @Shrenik0321
> > 
> > 
> > Alright, thanks. I'll be able to refer to that an write tests for future prs.
> 
> Great, can you write tests for this? Get some help with chatGPT. @Shrenik0321

yeah sure!

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/routes/profile.route.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import express from 'express';
+import passport from 'passport';
+import { getProfile } from '../controllers/profile.controller';
+
+const profileRouter = express.Router();
+
+profileRouter.get(
+  '/profile',
+  passport.authenticate('jwt', { session: false }),
```
[the remark this answers is not in the document]
Comment (Shrenik0321) [reply src/routes/profile.route.ts:9]: Sure!

File: src/services/auth.service.ts [hunk 5]
```diff
@@ -1,22 +1,28 @@
 import { dataSource } from '../configs/dbConfig';
-import { Request, Response } from 'express';
+import { Request, Response, NextFunction } from 'express';
 import jwt from 'jsonwebtoken';
 import bcrypt from 'bcrypt';
+import passport from 'passport';
 import Profile from '../entity/profile.entity';
 import { FindOneOptions } from 'typeorm';
 import { JWT_SECRET } from '../configs/envConfig';
+// import passport from 'passport';
```
Comment (anjula-sack) [inline src/services/auth.service.ts:9]: Let's remove this line

```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Review by anjula-sack — APPROVED** [review]
Good job! @Shrenik0321

**Comment by anjula-sack** [comment]
Fixes #10

**Pull request merged**